client/files.go

```diff
@@ -609,23 +614,29 @@ func (client *Client) Pull(opts *PullOptions) error {
 		return fmt.Errorf(`expected second field name to be "response", got %q`, responsePart.FormName())
 	}
 
-	// Process response metadata (see defaultRequester.Do)
+	return processPullResponsePart(responsePart)
+}
+
+// processPullResponsePart decodes the response part and extracts any errors
+func processPullResponsePart(part io.Reader) error {
 	var multipartResp response
-	if err := decodeInto(responsePart, &multipartResp); err != nil {
+	if err := decodeInto(part, &multipartResp)
+	err != nil {
```

Comment:
There's some funny formatting changes here, with the `err != nil {` split onto the next line (here and two places below). You can see this in the "Format" workflow failures here: https://github.com/canonical/pebble/actions/runs/20191040984/job/58022154319

Reply:
Yes, exactly. If you look at the diff here on GitHub you'll see that's what the original was.